Sources/Sentry/SentryBacktrace.cpp

```diff
@@ -103,16 +104,29 @@ namespace profiling {
 
     void
     enumerateBacktracesForAllThreads(const std::function<void(const Backtrace &)> &f,
-        const std::shared_ptr<ThreadMetadataCache> &cache)
+        const std::shared_ptr<ThreadMetadataCache> &cache,
+        const std::vector<std::unique_ptr<ThreadHandle>> &threads,
+        const std::unique_ptr<ThreadHandle> &currentThread)
     {
-        const auto pair = ThreadHandle::allExcludingCurrent();
-        for (const auto &thread : pair.first) {
+        if (currentThread == nullptr) {
+            SENTRY_PROF_LOG_WARN("Current thread handle (profiler's sampling thread) was null, "
+                                 "will not attempt to gather other threads' backtraces.");
+            return;
+        }
+
+        for (const auto &thread : threads) {
             Backtrace bt;
             // This one is probably safe to call while the thread is suspended, but
             // being conservative here in case the platform time functions take any
             // locks that we're not aware of.
             bt.absoluteTimestamp = getAbsoluteTime();
 
+            if (thread == nullptr) {
+                SENTRY_PROF_LOG_WARN(
+                    "Thread handle was null, will not attempt to gather its backtrace.");
+                continue;
+            }
```

Comment:
`l`: Any reason we can't put this at the beginning of the loop / before `Backtrace bt;` to avoid allocating the `Backtrace` struct?

Reply:
yep, i didn't look up enough. that's better for sure.